libnest/src/config_parser.rs

```diff
@@ -0,0 +1,146 @@
+//! Nest configuration file parsing implementation.
+//!
+//! All the code fetching the settings from the config file is here, in
+//! `ConfigParser::load_to_config`.
+//! If settings are added, this is where code must be added to fetch these new settings.
+//!
+//!
+
+extern crate toml;
+
+use std::fs::File;
+use std::io::BufReader;
+use std::io::prelude::Read;
+use std::io;
+use std::path::PathBuf;
+use std::fmt;
+
+use config::Config;
+use repository::{Mirror, Repository};
+
+pub(crate) enum ParseConfError {
+    Io(io::Error),
+    Deserialize(toml::de::Error),
+    Str(String),
+}
+
+impl fmt::Display for ParseConfError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            ParseConfError::Io(ref err) => write!(f, "{}", err),
+            ParseConfError::Deserialize(ref err) => write!(f, "{}", err),
+            ParseConfError::Str(ref err) => write!(f, "{}", err),
+        }
+    }
+}
+
+/// A struct holding the TOML main value
+#[derive(Debug)]
+pub(crate) struct ConfigParser {
+    toml: toml::value::Value,
+}
+
+impl ConfigParser {
+    /// Creates a ConfigParser instance from a TOML file
+    ///
+    /// self.toml is considered safe to cast to a table after this
+    /// `self.toml.as_table().unwrap();`
+    pub(crate) fn new(path: &str) -> Result<ConfigParser, ParseConfError> {
```

Comment:
I think a `Box<Error>` would be a bit less pain-in-the-ass to use. And we don't care about optimizations in fatal error-handling situations: it's already fucked up.

Reply:
(Oh and you should take an `&Path` instead of an `&str`)